src/main/java/htsjdk/tribble/util/ParsingUtils.java

```diff
@@ -90,35 +82,36 @@ public static InputStream openInputStream(String path)
         return openInputStream(path, null);
     }
 
+    static private final Set<String> URL_SCHEMES = new HashSet<>(Arrays.asList("http", "ftp", "https"));
+
     /**
      * open an input stream from the given path and wrap the raw byte stream with a wrapper if given
      *
      * the wrapper will only be applied to paths that are not http, https, ftp, or file, i.e. any {@link java.nio.file.Path}
      * using a custom filesystem plugin
-     * @param path a uri like string
+     * @param uri a uri like string
      * @param wrapper to wrap the input stream in, may be used to implement caching or prefetching, etc
-     * @return
-     * @throws IOException
+     * @return An inputStream appropriately created from uri and conditionally wrapped with wrapper (only in certain cases
+     * @throws IOException when stream cannot be opened against uri
      */
-    public static InputStream openInputStream(String path, Function<SeekableByteChannel, SeekableByteChannel> wrapper)
+    public static InputStream openInputStream(final String uri, final Function<SeekableByteChannel, SeekableByteChannel> wrapper)
             throws IOException {
 
+        // will be empty string if there's no scheme
+        final String scheme = Optional.ofNullable(IOUtil.getScheme(uri)).orElse("");
```

Comment:
`getScheme` doesn't return a null, there's no need for this complex dance.  It's also definitely preferable to just have a ternary expression that checks for null instead of creating a new nullable just to check if it's empty.

Reply:
look at getScheme:

```    *
     * @return  The scheme component of this URI,
     *          or {@code null} if the scheme is undefined
     */
```

Reply:
if I use a ternary, I need to either add a line of code (to save the result of getSceme) or call it twice....I'll do it if you insist, but I thought that this is a better solution.

Reply:
You're not calling uri.get scheme, you're calling your own version that you added, IOUtil.getScheme() which returns true/false.

Reply:
Ternary or not doesn't really matter, it's just kind of ugly to create a new wrapper object just to hide the null check.  I see your point about needing to call it twice or store it in a temp variable.  In any case, it feels like something is weird here.  You never actually use the scheme you get, you just check if it's empty.  Isn't calling `IOUtil.hasScheme`, which is what was originally done here correct?